pkg/sink/kafka/sarama_factory.go

```diff
@@ -14,44 +14,44 @@
 package kafka
 
 import (
-	"context"
-
 	"github.com/Shopify/sarama"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/pkg/errors"
 	"github.com/pingcap/tiflow/pkg/util"
 )
 
 type saramaFactory struct {
-	changefeedID    model.ChangeFeedID
-	brokerEndpoints []string
-	config          *sarama.Config
+	changefeedID model.ChangeFeedID
+	option       *Options
+	config       *sarama.Config
 }
 
 // NewSaramaFactory constructs a Factory with sarama implementation.
-func NewSaramaFactory(ctx context.Context,
+func NewSaramaFactory(
 	o *Options,
 	changefeedID model.ChangeFeedID,
 ) (Factory, error) {
-	saramaConfig, err := NewSaramaConfig(ctx, o)
+	saramaConfig, err := NewSaramaConfig(o)
 	if err != nil {
 		return nil, err
 	}
 	return &saramaFactory{
-		changefeedID:    changefeedID,
-		brokerEndpoints: o.BrokerEndpoints,
-		config:          saramaConfig,
+		changefeedID: changefeedID,
+		option:       o,
+		config:       saramaConfig,
 	}, nil
 }
 
 func (f *saramaFactory) AdminClient() (ClusterAdminClient, error) {
-	return newAdminClient(f.brokerEndpoints, f.config, f.changefeedID)
+	f.config.Producer.MaxMessageBytes = f.option.MaxMessageBytes
```

Comment:
Could you please add some tests for this code?

Reply:
Can we just apply it when we new the factory?

Reply:
after the factory created, it's option would be altered by `AdjustOptions` later.

Reply:
It's difficult to maintain it. If we change any options in `AdjustOptions `, then we need to update it there times. How about we create the config from options when we using it? Or pass the `f.config` into AdjustOptions?